src/lava/proc/lif/models.py

```diff
@@ -145,9 +157,151 @@ def run_spk(self):
         updated_volt = decayed_volt + self.u + self.effective_bias
         self.v[:] = np.clip(updated_volt, neg_voltage_limit, pos_voltage_limit)
 
-        # Spike when exceeds threshold
-        # ----------------------------
-        s_out = self.v >= self.effective_vth
+    def reset_voltage(self, spike_vector: np.ndarray):
+        """Voltage reset behaviour. This can differ for different neuron
+            models.
+        """
+        self.v[spike_vector] = 0
+
+    def run_spk(self):
+        """The run function that performs the actual computation during
+        execution orchestrated by a PyLoihiProcessModel using the
+        LoihiProtocol
+        """
+        # Receive synaptic input
+        a_in_data = self.a_in.recv()
+
+        # Compute effective bias and threshold only once, not every time-step
+        if not self.isbiasscaled:
+            self.scale_bias()
+
+        if not self.isthrscaled:
+            self.scale_threshold()
```

Comment:
Why can't these two scalings be done on __init__?  as it is, you still have these two checks at every timestep during runtime.

Reply:
In \_\_init\_\_(), the variables are still LavaPyTypes. They get built into numpy arrays only after a ProcessModel instance is created.